lldb/source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -2095,6 +2095,50 @@ void SwiftLanguageRuntime::DidFinishExecutingUserExpression(
 
 bool SwiftLanguageRuntime::IsABIStable() { FORWARD(IsABIStable); }
 
+struct AsyncUnwindRegisters {
+  uint32_t async_ctx_regnum;
+  uint32_t fp_regnum;
+  uint32_t pc_regnum;
+  uint32_t dummy_regnum;
+};
+
+static llvm::Optional<AsyncUnwindRegisters>
+GetAsyncUnwindRegisters(llvm::Triple::ArchType triple) {
+  switch (triple) {
+  case llvm::Triple::x86_64:
+    return (AsyncUnwindRegisters){
+        .async_ctx_regnum = dwarf_r14_x86_64,
+        .fp_regnum = dwarf_rbp_x86_64,
+        .pc_regnum = dwarf_rip_x86_64,
+        .dummy_regnum = dwarf_r15_x86_64,
+    };
+  case llvm::Triple::aarch64:
+    return (AsyncUnwindRegisters){
+        .async_ctx_regnum = arm64_dwarf::x22,
+        .fp_regnum = arm64_dwarf::fp,
+        .pc_regnum = arm64_dwarf::pc,
+        .dummy_regnum = arm64_dwarf::x23,
+    };
+  default:
+    assert(false && "swift async supports only x86_64 and arm64");
```

Comment:
I think this may actually a proper use-case for lldb_assert()?

Reply:
That would be a non-NFC change. Currently `GetRuntimeUnwindPlan()` returns a null plan, which I think is better than crashing. It'd be better to show an OS backtrace instead of crashing.